tests/parametric/test_span_links.py

```diff
@@ -0,0 +1,189 @@
+        link = span_links[0]
+        assert link.get("span_id") == root.get("span_id")
+        assert link.get("trace_id") == root.get("trace_id")
+        assert (link.get("trace_id_high") or 0) == 0
+        assert link["attributes"].get("foo") == "bar"
+        assert link["attributes"].get("array.0") == "a"
+        assert link["attributes"].get("array.1") == "b"
+        assert link["attributes"].get("array.2") == "c"
+        assert (link.get("flags") or 0) == 0
+
+    def test_span_link_from_distributed_datadog_headers(self, test_agent, test_library):
+        """Properly inject datadog distributed tracing information into span links.
+        Testing the conversion of x-datadog-* headers to tracestate for
+        representation in span links.
+        """
+        with test_library:
+            with test_library.start_span(
+                "root",
+                links=[Link(parent_id=0, attributes={"foo": "bar"})],
+                http_headers=[
+                    ["x-datadog-trace-id", "1234567890"],
+                    ["x-datadog-parent-id", "9876543210"],
+                    ["x-datadog-sampling-priority", "2"],
+                    ["x-datadog-origin", "synthetics"],
+                    ["x-datadog-tags", "_dd.p.dm=-4,_dd.p.tid=0000000000000010"],
+                ],
+            ):
+                pass
+
+        span = get_span(test_agent)
+        assert span.get("trace_id") != 1234567890
+        assert span_has_no_parent(span)
+        assert span["meta"].get(ORIGIN) is None
+        assert span["meta"].get("_dd.p.dm") != "-4"
+        assert span["metrics"].get(SAMPLING_PRIORITY_KEY) != 2
+
+        assert span.get("span_links") is not None
+
+        span_links = span["span_links"]
+        assert len(span_links) == 1
+        link = span_links[0]
+        assert link.get("span_id") == 9876543210
+        assert link.get("trace_id") == 1234567890
+        assert link.get("trace_id_high") == 16
+
+        assert link.get("tracestate") is not None
+        tracestateArr = link["tracestate"].split(",")
+        assert len(tracestateArr) == 1 and tracestateArr[0].startswith("dd=")
+        tracestateDD = tracestateArr[0][3:].split(";")
+        assert len(tracestateDD) == 3
+        assert "o:synthetics" in tracestateDD
+        assert "s:2" in tracestateDD
+        assert "t.dm:-4" in tracestateDD
+
+        assert (link.get("flags") or 0) == 0
+        assert len(link.get("attributes")) == 2
+        assert link["attributes"].get("foo") == "bar"
+        assert link["attributes"].get("_dd.p.dm") == "-4"
+
+    def test_span_link_from_distributed_w3c_headers(self, test_agent, test_library):
+        """Properly inject w3c distributed tracing information into span links.
+        This mostly tests that the injected tracestate and flags are accurate.
+        """
+        with test_library:
+            with test_library.start_span(
+                "root",
+                links=[Link(parent_id=0)],
```

Comment:
Instead of using Link(parent_id=0) to tell the tracer to set this as a span link (not span parent), can we add a separate flag to the start_span method call?

Reply:
Any specific reason why? Like that's what I initially thought to do, then decided that it'd be better to not have one or the other parameter completely meaningless depending on whether the one thing is non-zero or the other being true, removing all ambiguity hereby.

Reply:
To me the `parent_id=0` introduces ambiguity 😆 Once I realized it had a special meaning, it was fine, but I thought a separate flag would be more intuitive as a branching condition to for parent vs span link

Reply:
Yes, it's really "when no explicit parent_id is set", ...

Reply:
As far as this comment thread goes, it's resolved to me. Anything outstanding for you @PerfectSlayer ?

Reply:
So, let me check if I understand everything. 

If it there is a link with `parent_id` not set or set to `0`, then we should have a look at the distributed headers attributes for the `StartSpan` message to create a span link?
Moreover, if there are attributes, we should use both distributed headers and attributes to build the span links?

And what about the default context extraction behavior? Because using distributed headers will already trigger span link creation if playing with distributed headers 🤔  

To me, if feels way to complicated / ambiguous. Instead, I would have:

```protobuf
message SpanLink {
  oneof from {
      uint64 parent_id = 1;
      DistributedHTTPHeaders http_headers = 2;
    }
  Attributes attributes = 3;
}
```

Reply:
I changed it to your proposal in the proto.